analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -125,8 +125,11 @@ public bool HasMatchingScope(DiagnosticDescriptor descriptor)
             descriptor.CustomTags.Contains(tag);
     }
 
-    private bool ShouldAnalyzeFile(SonarLintXmlReader sonarLintXml, string filePath) =>
-        ProjectConfiguration().ProjectType != ProjectType.Unknown // Not SonarLint context, NuGet or Scanner <= 5.0
+    private bool FileIsIncludedInAnalysis(SonarLintXmlReader sonarLintXml, string filePath) =>
+        // If ProjectType != 'Unknown' we are in S4NET context and all files are analyzed.
+        // If ProjectType == 'Unknown' then we are in SonarLint or NuGet context and
+        // we need to check if the file has been excluded from analysis through SonarLint.xml.
```

Comment:
This fits on a single line.
```suggestion
        // If ProjectType == 'Unknown' then we are in SonarLint or NuGet context and we need to check if the file has been excluded from analysis through SonarLint.xml.
```